src/main/java/jenkins/plugins/slack/HttpClient.java

```diff
@@ -50,6 +51,17 @@ public static CloseableHttpClient getCloseableHttpClient(ProxyConfiguration prox
         }
         return clientBuilder.build();
     }
+    
+    private static boolean isNoProxyHost(String host, String noProxyHost) {
+        if (host!=null && noProxyHost!=null) {
+            for (Pattern p : ProxyConfiguration.getNoProxyHostPatterns(noProxyHost)) {
```

Comment:
Not sure how the proxy host matcher works but surely this won't match `https://{teamDomain}.slack.com`

https://github.com/jenkinsci/slack-plugin/blob/610381174a22d9c3a4aac660bd8df01b7b9a860c/src/main/java/jenkins/plugins/slack/StandardSlackService.java#L198-L205